usaspending_api/common/threaded_data_loader.py

```diff
@@ -0,0 +1,194 @@
+            thread.start()
+            pool.append(thread)
+
+        with open(filepath) as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                row_queue.put(row)
+
+        while not row_queue.empty():
+            pass
+
+        self.exit_flag = True
+
+        for thread in pool:
+            thread.join()
+
+        if self.post_process_function is not None:
+            self.post_process_function()
+
+        end_time = time.time()
+        elapsed = end_time - start_time
+        print('Finished processing at ' + str(end_time) + '; Elapsed: ' + str(elapsed))
+
+
+class DataLoaderThread(threading.Thread):
+    def __init__(self, name, data_queue, loader):
+        threading.Thread.__init__(self)
+        self.name = name
+        self.data_queue = data_queue
+        self.loader = loader
+
+    def run(self):
+        print("Starting " + self.name)
+        self.process_data()
+        print("Exiting " + self.name)
+
+    def process_data(self):
+        # Make sure we weren't flagged to exit
+        while not self.loader.exit_flag:
+            if not self.data_queue.empty():
+                row = self.data_queue.get()
+                # Grab the collision field
+                update = False
+                collision_instance = None
+                model_instance = None
+                if self.loader.collision_field is not None:
+                    model_collision_field = self.loader.collision_field
+                    data_collision_field = model_collision_field
+                    # If it's a long label, we need to look it up to map to data
+                    if model_collision_field in settings.LONG_TO_TERSE_LABELS:
+                        data_collision_field = settings.LONG_TO_TERSE_LABELS[model_collision_field]
+                    query_object = Q({model_collision_field: row[data_collision_field]})
+                    collision_instance = self.loader.model_class.objects.filter(query_object).first()
+
+                    if collision_instance is not None:
+                        behavior = self.loader.collision_behavior
+                        if behavior is "delete":  # Delete the row from the data store and load new row
+                            collision_instnace.delete()
+                        if behavior is "update":  # Update the row in the data store with new values from the CSV
+                            update = True
+                        if behavior is "skip":  # Skip the row
+                            continue
+                        if behavior is "skip_and_complain":  # Log a warning and skip the row
+                            print('Hit a collision on row %s' % (row))
+                        if behavior is "die":  # Raise an exception and cease execution
+                            raise Exception("Hit collision on row %s" % (row))
+
+                if not update:
+                    model_instance = self.loader.model_class()
```

Comment:
The only time update is true is when the collision method is 'update' and there was a collision - there's no actual overwriting here. Either the model_instance to load data into is the collision_instance (in the case of an update) or it's brand new. Every other case either: skips the row, deletes the collision, or throws an error and dies. 

(Although this did make me notice I skipped a 'continue' statement)

If this is still unclear let me know. Since we're switching to the default collision_method being update, it should be a bit clearer